File: Python/person.py
class Person:
  def __init__(self, name):
    super().__init__();
    self.name = name;
  def __str__(self):
    return super().__str__() + f" Person {self.name}";

class Student(Person):
  def __init__(self, name, university):
    super().__init__(name);
    self.university = university;
  def __str__(self):
    return super().__str__() + f" Student {self.university}";

class Employee(Person):
  def __init__(self, name, company):
    super().__init__(name);
    self.company = company;
  def __str__(self):
    return super().__str__() + f" Employee {self.company}";

p = Person("Adam");
s = Student("Bertil", "Umea");
e = Employee("Ceasar", "Volvo");

print(p);
print(s);
print(e);

File: Ruby/person.rb
class Person 
  String :m_name

  def initialize name:
    @m_name = name
  end

  public def to_s
    "Person #{@m_name}"
  end
end

class Student < Person
  String :m_university

  def initialize name:, university:
    super name: name
    @m_university = university
  end

  public def to_s
    "#{super} Student #{@m_university}"
  end
end

class Employee < Person
  String :m_company

  def initialize name:, company:
    super name: name
    @m_company = company
  end

  public def to_s
    "#{super} Employee #{@m_company}"
  end
end

p = Person.new name: "Adam"
puts p

s = Student.new name: "Bertil", university: "Umea"
puts s

e = Employee.new name: "Ceasar", company: "Volvo"
print e
